glommio/src/executor/stall.rs

```diff
@@ -0,0 +1,283 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+//
+#![allow(dead_code)]
+
+use nix::sys;
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread::JoinHandle,
+    time::{Duration, Instant},
+};
+
+#[cfg(feature = "stall-detection")]
+thread_local! {
```

Comment:
TLS is a bad choice. This should be a member of the reactor.

Reply:
Good point. They should be members of the `StallDetector` struct that I can access through the `LocalExecutor`. Will update.